Cargo.toml

```diff
@@ -17,9 +17,18 @@ codecov = { repository = "https://github.com/c0dearm/sharks" }
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
+[lib]
+crate-type = ["cdylib", "rlib"]
```

Comment:
I've been thinking, maybe we can split the WASM bindings into a different crate, such as `sharks-wasm`, and put everything in a Cargo workspace. I think it could be cleaner and more maintainable. As suggested here: https://stackoverflow.com/questions/40514793/how-can-i-use-cargo-to-build-a-library-as-both-an-rlib-and-a-dylib-but-with-diff

What do you think?

Reply:
With my experience with WASM bindings, the fact that cargo doesn't support conditionnal lib type is really a pain and in fact, splitting it into another crate could be a good idea. 

Also, the wasm version doesn't need to be uploaded on crates.io, since no one should use it from rust, but directly to NPM under the name `sharks`.

What's important is to make sure the `wasm` part gets run in the pipeline to make sure a change in the main crate doesn't break it.

Reply:
With my experience with WASM bindings, the fact that cargo doesn't support conditionnal lib type is really a pain and in fact, splitting it into another crate could be a good idea.

Also, the wasm version doesn't need to be uploaded on crates.io, since no one should use it from rust, but directly to NPM under the name sharks.

What's important is to make sure the wasm part gets run in the pipeline to make sure a change in the main crate doesn't break it.

Reply:
Fantastic, when I have some time between today and tomorrow I will refactor the project into a Cargo workspace and put the WASM bindings in it, along the original crate.

Reply:
Good :)